test/test_torch.py

```diff
@@ -13367,6 +13367,35 @@ def test_cat_big(self, device):
         result = torch.cat(concat_list)
         self.assertEqual(result.size(0), SIZE1 + SIZE2)
 
+    @onlyCUDA
+    def test_atomic_add(self, device):
+        # todo: we should probably expose a dtype.is_signed()
+        def is_signed(dtype):
+            return torch.is_signed(torch.tensor(0, dtype=dtype))
+        # https://github.com/pytorch/pytorch/issues/29153
+        for dtype in torch.testing.get_all_math_dtypes(device):
+            size = [5, 5]
+            if dtype.is_floating_point:
+                tensor = torch.rand(size, dtype=dtype, device=device)
+            if is_signed(dtype):
+                tensor = torch.randint(-5, 15, size, dtype=dtype, device=device)
+            else:
+                tensor = torch.randint(0, 10, size, dtype=dtype, device=device)
+
+            # index_add calls atomicAdd on cuda.
+            empty = torch.zeros(size, dtype=dtype, device=device)
+            added = empty.index_add(0, torch.arange(0, size[0], dtype=torch.long, device=device), tensor)
+            self.assertEqual(added.sum(), tensor.sum())
+
+            # sparse.to_dense() on non-coalesced tensors calls index_add, which was where
```

Comment:
Perhaps this should be a different test case (and should be placed inside `test_sparse`)?

Reply:
it's probably not really necessary after I add the next stacked diff anyway. it was just my original test before I drilled deeper.